pkg/operator/metrics/metrics.go

```diff
@@ -0,0 +1,62 @@
+package metrics
+
+import (
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"sync"
+	"time"
+)
+
+// ServiceCAMetrics implements metrics gathering for this component.
+type ServiceCAMetrics struct {
+	lock       sync.Mutex
```

Comment:
There should be no need to lock any of this. The metrics are set atomically and you should just register the metrics directly into the metrics registry.

Reply:
Metrics in this instance are being written by one goroutine (the operator worker) and read by another (the server exposing the metrics for collection). That suggests synchronization to me. Is there another strategy for ensuring safe concurrent access to the state variables that you would suggest?

Reply:
Yes just create and register the metrics once, the registration process is already concurrency safe.

Reply:
I'm not following. Can you point me to code that is ensuring that concurrent access to the state variables will be synchronized?

Reply:
Look at this example app for example https://github.com/brancz/prometheus-example-app/blob/master/main.go

Reply:
@brancz Thanks for the review. I understand that mutex is not required if we are only using one metric because of the atomicity. But if there are multiple metrics that are inter-dependent, don't we need a mutex in that case?

Reply:
Gotcha. As you were trying to say, `Gauge.Set` ensures synchronization by setting the value atomically.

Reply:
Okay! My page isn't refreshed and I didn't see your conversation.

Reply:
But my question is still open?